### PR TITLE
Continue on `contrib/scalar`

### PR DESCRIPTION
**Pull request by dscho**
The first regression test _almost_ works. I have to take off now, but the next step would be to figure out a way to verify that a promised object is not present locally yet.

### CODE REVIEW

File: contrib/scalar/scalar.c [hunk 1]
```diff
@@ -18,12 +19,13 @@ static char *scalar_executable_path;
 
 static int run_git(const char *dir, const char *arg, ...)
 {
-	struct strvec argv;
+	struct strvec argv = STRVEC_INIT;
```
Comment (derrickstolee) [inline contrib/scalar/scalar.c:24]: Whoops! Thanks!

**Review by derrickstolee — APPROVED** [review]
I'm not planning on doing work on this today, so you can continue building in this PR if you want. Approving to unblock a merge when you're ready.

File: contrib/scalar/scalar.c (hunk 1 above)
Comment (dscho) [reply contrib/scalar/scalar.c:24]: That was me, not you ;-)

File: contrib/scalar/scalar.c [hunk 2]
```diff
@@ -346,9 +348,181 @@ static int cmd_clone(int argc, const char **argv)
...
+		goto diagnose_cleanup;
+
+	/*
+	 * TODO: add more stuff:
+	 * disk space info
+	 * LogDirectoryEnumeration(...DotGit.Objects.Root), ScalarConstants.DotGit.Objects.Pack.Root, "packs-local.txt");
+	 * LogLooseObjectCount(...DotGit.Objects.Root), ScalarConstants.DotGit.Objects.Root, "objects-local.txt");
+	 *
+	 * CopyLocalCacheData(archiveFolderPath, gitObjectsRoot);
+	 */
```
Comment (derrickstolee) [inline contrib/scalar/scalar.c:507]: Thanks for commenting here. These are my favorite things to inspect in diagnose logs. We should be able to collect this data for each odb. It doesn't need to use the same structure as before.

Comment (dscho) [reply contrib/scalar/scalar.c:507]: Right, I figured. They were also the least trivial to implement, that's why I left them for later ;-)

**Pull request merged**